src/jobrunner.rs

```diff
@@ -40,15 +43,16 @@ struct JobRunner {
 
 impl JobRunner {
     fn new(snare: Arc<Snare>) -> Result<Self, Box<dyn Error>> {
-        assert!(snare.config.maxjobs < std::usize::MAX);
-        let mut running = Vec::with_capacity(snare.config.maxjobs);
-        running.resize_with(snare.config.maxjobs, || None);
-        let mut pollfds = Vec::with_capacity(snare.config.maxjobs * 2 + 1);
-        pollfds.resize_with(snare.config.maxjobs * 2 + 1, || {
-            PollFd::new(-1, PollFlags::empty())
-        });
+        // If the unwrap() on the lock fails, the other thread has paniced.
+        let maxjobs = snare.config.lock().unwrap().maxjobs;
+        assert!(maxjobs < std::usize::MAX);
```

Comment:
I'm not sure this is a meaningful assertion. Since both arguments are `usize` it's equivalent to:
```
assert!(maxjobs != std::usize::MAX);
```

Is that being used as a boundary condition elsewhere perhaps?

Reply:
Actually, this should be `assert!(maxjobs < ((std::usize::MAX - 1) / 2)`! Long story, but it's basically about the amount of pollfds we create. In practise, of course, we're probably not going to have enough RAM for this to ever be an issue.

Fixed in https://github.com/softdevteam/snare/pull/15/commits/5ba7c259a8e0c501cdced7f99020ca1efe9bc2c1.

Reply:
OK, that looks more valid at least.

I missed why the `/ 2`, but if the story is really long, I'll trust you ;)

Reply:
Each job has stderr/stdout pipes which is the `*` (or `/)` 2.

Reply:
ok